File: client/src/api/market.ts
import { apiFetch } from '@/api/client'

export interface IndexQuote {
  symbol: string
  name: string
  price: number
  change: number
  change_percent: number
}

export interface QuoteSnapshot {
  symbol: string
  price: number
  volume: number
  timestamp: number
  sparkline: number[]
}

export interface SearchResult {
  symbol: string
  description: string
  type: string
}

export interface MarketStatus {
  is_open: boolean
  holiday: string | null
}

export function fetchMarketStatus(): Promise<MarketStatus> {
  return apiFetch('/market/status')
}

export function fetchIndices(): Promise<IndexQuote[]> {
  return apiFetch('/market/indices')
}

export function fetchLatestQuotes(symbols: string[]): Promise<QuoteSnapshot[]> {
  return apiFetch(`/quotes/latest?symbols=${symbols.join(',')}`)
}

export function searchSymbols(query: string): Promise<SearchResult[]> {
  return apiFetch(`/search?q=${encodeURIComponent(query)}`)
}

File: server/app/schemas/market.py
from pydantic import BaseModel


class IndexQuote(BaseModel):
    symbol: str
    name: str
    price: float
    change: float
    change_percent: float


class QuoteSnapshot(BaseModel):
    symbol: str
    price: float
    volume: int = 0
    timestamp: int = 0
    sparkline: list[float] = []


class SymbolSearchResult(BaseModel):
    symbol: str
    description: str
    type: str


class MarketStatus(BaseModel):
    is_open: bool
    holiday: str | None = None
